File: Dev/C++/Mascaret/src/VEHA/Kernel/LiteralBoolean.cpp
#include "VEHA/Kernel/LiteralBoolean.h"
#include "VEHA/Kernel/PrimitiveType.h"
#include "MascaretApplication.h"
#include "VEHA/Kernel/Model.h"

namespace VEHA
{

LiteralBoolean::LiteralBoolean(bool b)
:LiteralSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("boolean")), _value(b)
{
}
LiteralBoolean::LiteralBoolean(const string & str)
:LiteralSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("boolean")),_value(false)
{
	setValueFromString(str);
}

LiteralBoolean::~LiteralBoolean()
{
}

bool LiteralBoolean::setValueFromString(const string& value)
{
	if (value == "TRUE" || value == "true") _value = true;
	else if (value == "FALSE" || value == "false") _value = false;
	else return false;
	return true;
}

string LiteralBoolean::getStringFromValue() const
{
	return (_value?"true":"false");
}
LiteralBoolean::operator string () const
        		{
	return  boost::lexical_cast<std::string>(_value);
        		}

LiteralBoolean::operator bool () const
        		{
	return _value;
        		}
//const ValueSpecification& LiteralBoolean::operator= (bool value)
//{
//	_value=value;
//	return *this;
//}
shared_ptr<ValueSpecification> LiteralBoolean::clone() const
{
	LiteralBoolean * vs=new LiteralBoolean();
	vs->CLASSTYPE=CLASSTYPE;
	vs->_type=_type;
	vs->_value=_value;
	return shared_ptr<ValueSpecification>(vs);
}
//const ValueSpecification& LiteralBoolean::operator= (const ValueSpecification&vs)
//{
//	_value=(bool)vs;
//	return *this;
//}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Kernel/LiteralBoolean.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class LiteralBoolean : LiteralSpecification
    {

        protected bool bValue;
        public bool BValue
        {
            get { return bValue; }
            set { bValue = value; }
        }

        public bool setValueFromString(string sValue)
        {
            if (sValue.ToLower() == "true") bValue = true;
            else if (sValue.ToLower() == "false") bValue = false;
            else return false;
            return true;
        }

        public override string getStringFromValue()
        {
            return (bValue ? "true" : "false");
        }

        public override bool getBoolFromValue()
        {
            return bValue;
        }

        public LiteralBoolean()
            : base(MascaretApplication.Instance.Model.getBasicType("boolean"))
        {
        }

        public LiteralBoolean(bool b)
            : base(MascaretApplication.Instance.Model.getBasicType("boolean"))
        {
            bValue = b;
        }

        public LiteralBoolean(string svalue)
            : base(MascaretApplication.Instance.Model.getBasicType("boolean"))
        {
            bValue = false;
            setValueFromString(svalue);
        }

        public override ValueSpecification clone()
        {
            LiteralBoolean vs = new LiteralBoolean("");
            vs.CLASSTYPE = CLASSTYPE;
            vs.Type = this.Type;
            vs.bValue = bValue;
            return ((ValueSpecification)vs);
        }
    }
}